crates/fig_os_shim/src/fs.rs

```diff
@@ -608,4 +822,58 @@ mod tests {
         fs.create_new("my_file.txt").await.unwrap();
         assert!(fs.create_new("my_file.txt").await.is_err());
     }
+
+    #[tokio::test]
+    #[cfg(windows)]
+    async fn test_unified_symlink_windows() {
+        let dir = tempfile::tempdir().unwrap();
+        let fs = Fs::new();
+
+        // Create a test file
+        let file_path = dir.path().join("test_file.txt");
+        fs.write(&file_path, "test content").await.unwrap();
+
+        // Create a test directory
+        let dir_path = dir.path().join("test_dir");
+        fs.create_dir(&dir_path).await.unwrap();
+
+        // Test symlink to file
+        let file_link_path = dir.path().join("file_link");
+        match fs.symlink(&file_path, &file_link_path).await {
+            Ok(_) => {
+                // If we have permission to create symlinks, run the full test
+                assert_eq!(fs.read_to_string(&file_link_path).await.unwrap(), "test content");
+
+                // Test symlink to directory
+                let dir_link_path = dir.path().join("dir_link");
+                fs.symlink(&dir_path, &dir_link_path).await.unwrap();
+                assert!(fs.try_exists(&dir_link_path).await.unwrap());
+
+                // Test symlink_sync to file
+                let file_link_sync_path = dir.path().join("file_link_sync");
+                fs.symlink_sync(&file_path, &file_link_sync_path).unwrap();
+                assert_eq!(fs.read_to_string(&file_link_sync_path).await.unwrap(), "test content");
+
+                // Test symlink_sync to directory
+                let dir_link_sync_path = dir.path().join("dir_link_sync");
+                fs.symlink_sync(&dir_path, &dir_link_sync_path).unwrap();
+                assert!(fs.try_exists(&dir_link_sync_path).await.unwrap());
+
+                // Clean up
+                fs.remove_file(&file_link_path).await.unwrap();
+                fs.remove_file(&file_link_sync_path).await.unwrap();
+                fs.remove_dir_all(&dir_link_path).await.unwrap();
+                fs.remove_dir_all(&dir_link_sync_path).await.unwrap();
+            },
+            Err(e) if e.kind() == std::io::ErrorKind::PermissionDenied || e.raw_os_error() == Some(1314) => {
```

Comment:
I catch permissions issues for tests so that VSCode doesn't need to be run as an admin but we can change this to force it. Another option is to just ensure that there are required permissions when pipeline actions run.